docs/sphinx/api/builtin_ops.rst

```diff
@@ -0,0 +1,255 @@
+Built-in Quantum Operations
```

Comment:
Pedantically, I'm not sure we should really call them _built-in_. CUDA Quantum operations are defined as C++ functions, etc. just like any other C++ library. (You can compile a cudaq program with standard clang++ and the correct command-line switches and libraries.)

Reply:
FWIW, a built-in can refer to some special additional bits that a compiler recognizes but that are not particularly a part of the language standard or definition. For example, a type like `__builtin__int512` might be recognized by a compiler as a built-in type since it might be useful on some particular target architecture and there is no standard conforming way of declaring such an extra-long word, etc.

Cudaq doesn't extend C++ in this way, nor does the nvq++ compiler recognize special sauce features. The closest thing to this would be the quantum attribute. But that can be ignored by conforming C++ compilers, so Cudaq code can be compiled and execute in an emulation mode, dropping the attribute on the floor.

My suggestion would be not to use the "built-in" adjective. "Quantum Operations" seems sufficient to me. These would be the "gates" that can be used to compose "user-defined quantum operations", a.k.a. quantum kernels/functions.

Reply:
Alright, I removed any use of "built-in". I am not sure if the page is very helpful regarding the existence of a template for some of the quantum kernel but not others (why would an operation swap be any different than e.g. my custom flip_and_swap kernel?), but let's see if that causes any confusion - a topic for another PR; I don't think this PR is making that much more or less clear than it is now.